### PR TITLE
Sv 7288/delete date

### PR DESCRIPTION
**Pull request by mandymichel**
This creates a new property canDelete that defaults as true. If there is a date set, the user can delete it by clicking the close icon and it fires the same event, only with a null value. On setDate, I messed up the types, so guidance on that would be nice.

### CODE REVIEW

File: packages/date/src/chameleon-date.ts [hunk 1]
```diff
@@ -351,21 +365,22 @@ export default class ChameleonDate extends LitElement {
     this.renderedDate = new Date(date);
   }
 
-  private async setDate(e: MouseEvent): Promise<void> {
+  private async setDate(e: any): Promise<void> {
```
Comment (ryuhhnn) [inline packages/date/src/chameleon-date.ts:368]: Can you add this event back in? I don't want to regress from having strongly typed elements towards having `any`s sneak their way in.

File: packages/date/src/chameleon-date.ts [hunk 2]
```diff
@@ -351,21 +365,22 @@ export default class ChameleonDate extends LitElement {
...
     this.touched = true;
-    const date = (<DateSelectTarget>e.target)!.value;
-    this.date = date;
-    this.active = false;
-
-    this.requestUpdate();
-    await this.updateComplete;
+    if (e && e.target && e.target.value) {
```
Comment (ryuhhnn) [inline packages/date/src/chameleon-date.ts:370]: Can we typescriptify this by saying `e.target?.value`?

File: packages/date/src/chameleon-date.ts [hunk 3]
```diff
@@ -134,16 +137,35 @@ export default class ChameleonDate extends LitElement {
...
+            class="delete"
+            @click="${this.delete}"
+            >${this.closeIcon}
+          </chameleon-button>`
         : nothing}
     `;
   }
 
+  delete() {
+    this.touched = true;
```
Comment (ryuhhnn) [inline packages/date/src/chameleon-date.ts:164]: Is this redundant? If the user enters a date, the element will have already been set to `this.touched = true`

File: packages/date/src/chameleon-date.ts [hunk 4]
```diff
@@ -134,16 +137,35 @@ export default class ChameleonDate extends LitElement {
...
+    if (this.value && this.value.length) this.value = "";
```
Comment (ryuhhnn) [inline packages/date/src/chameleon-date.ts:165]: Can we replace this with `this.value?.length`?

File: packages/date/src/chameleon-date.ts (hunk 1 above)
Comment (mandymichel) [reply packages/date/src/chameleon-date.ts:368]: If I put the MouseEvent back in, it won't let me pass in null. I could make 2 different methods for if there is a value or not, but was hoping to use the some method for both. Ideas for the type?

File: packages/date/src/chameleon-date.ts (hunk 3 above)
Comment (mandymichel) [reply packages/date/src/chameleon-date.ts:164]: Yes

**Pull request merged**